opentelemetry-sdk/src/trace/span.rs

```diff
@@ -566,6 +572,58 @@ mod tests {
         );
     }
 
+    #[test]
+    fn exceed_span_attributes_value_length_limit() {
+        let max_attribute_value_length = 3;
+        let exporter = NoopSpanExporter::new();
+        let provider_builder = crate::trace::SdkTracerProvider::builder()
+            .with_simple_exporter(exporter)
+            .with_max_attribute_value_length(max_attribute_value_length);
+        let provider = provider_builder.build();
+        let tracer = provider.tracer("opentelemetry-test");
+
+        let initial_attributes = vec![
+            KeyValue::new("first", String::from("test data")),
+            KeyValue::new(
+                "second",
+                Value::Array(
+                    [
+                        StringValue::from("test data"),
+                        StringValue::from("test data, once again"),
+                    ]
+                    .to_vec()
+                    .into(),
+                ),
+            ),
+        ];
+        let span_builder = SpanBuilder::from_name("test_span").with_attributes(initial_attributes);
+
+        let mut span = tracer.build(span_builder);
+        span.set_attribute(KeyValue::new("third", "test data, after span builder"));
+
+        span.with_data(|data| {
+            assert_eq!(
```

Comment:
instead of picking attributes at each index(0,1,..), could we do a general validation - for each string or array of string, validate that length is <max_attribute_value_length?